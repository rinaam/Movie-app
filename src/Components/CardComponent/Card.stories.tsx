import React from "react";
import { Story, Meta } from "@storybook/react";

import CardComponent, { ICardProps } from "./CardComponent";

export default {
  title: "Card",
  component: CardComponent,
} as Meta;

const Template: Story<ICardProps> = (args) => <CardComponent {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  grade: 5,
  imgUrl: `https://image.tmdb.org/t/p/w342/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg`,
  title: "Movie Title Movie Title Movie Title",
  content: "Language: en",
  titleSufix: "2018",
};
