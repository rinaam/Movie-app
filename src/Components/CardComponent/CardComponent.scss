@use "sass:map";
@import "../../Styles/variables.scss";
@import "../../Styles/breakpoints.scss";

.card {
  box-shadow: map.get($box-shadow, "elevation");
  border-radius: 4px;
  width: 290px;
  background-color: $white;
  padding: map.get($spacing, "s");
  margin: map.get($spacing, "xs");
  display: inline-block;
  cursor: pointer;
  @include breakpoint(smallDevices) {
    margin: map.get($spacing, "m");
    width: 350px;
  }
  &-header {
    position: relative;
    height: 500px;
    &-grade {
      position: absolute;
      top: -15px;
      right: -15px;
      background-color: $white-smoke;
      border-radius: 100%;
      height: 50px;
      width: 50px;
      display: flex;
      justify-content: center;
      align-items: center;
    }
    &-image {
      width: 100%;
      height: 100%;

      background: {
        repeat: no-repeat;
        size: cover;
        position: center;
      }
    }
  }
  &-data {
    line-height: 1.5;
    padding-top: map.get($spacing, "m");
    &-title {
      display: flex;
      :first-child {
        margin-right: map.get($spacing, "xs");
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
      }
    }
  }
}
