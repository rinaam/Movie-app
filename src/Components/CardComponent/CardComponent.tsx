import React from "react";
import "./CardComponent.scss";

import ImageComponent from "../ImageComponent/ImageComponent";

export interface ICardProps {
  grade: number;
  imgUrl: string;
  title: string;
  content: string;
  titleSufix: string | number;
  onClick: () => void;
}

const CardComponent: React.FC<ICardProps> = ({
  grade,
  imgUrl,
  title,
  content,
  titleSufix,
  onClick,
}) => (
  <div className="card" onClick={onClick}>
    <div className="card-header">
      <ImageComponent
        alt={title}
        imgUrl={imgUrl}
        useBackground
        className="card-header-image"
      />

      <p className="card-header-grade">{grade}</p>
    </div>
    <div className="card-data">
      <div className="card-data-title">
        <p>{title}</p>
        <p>({titleSufix})</p>
      </div>

      <p> {content}</p>
    </div>
  </div>
);

export default CardComponent;
