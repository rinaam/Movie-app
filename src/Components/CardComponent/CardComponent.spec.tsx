import { create } from "react-test-renderer";
import CardComponent from "./CardComponent";
import React from "react";

describe("CardComponent", () => {
  it("should render CardComponent", () => {
    const wrapper = create(
      <CardComponent
        grade={5}
        imgUrl="image"
        title="Movie Title"
        content="Language: en"
        titleSufix="2018"
        onClick={jest.fn}
      />
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
