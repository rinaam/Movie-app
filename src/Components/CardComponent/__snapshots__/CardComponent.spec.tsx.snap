// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent should render CardComponent 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <div
    className="card-header"
  >
    <div
      className="card-header-image"
      style={
        Object {
          "backgroundImage": "url(\\"image\\")",
        }
      }
    />
    <p
      className="card-header-grade"
    >
      5
    </p>
  </div>
  <div
    className="card-data"
  >
    <div
      className="card-data-title"
    >
      <p>
        Movie Title
      </p>
      <p>
        (
        2018
        )
      </p>
    </div>
    <p>
       
      Language: en
    </p>
  </div>
</div>
`;
